src/Web3/Eth.elm

```diff
@@ -30,21 +109,140 @@ getBlockNumber =
         }
 
 
-getBlock : Int -> Task Error Block
-getBlock blockNum =
+getBalance : Address -> Task Error BigInt
+getBalance address =
+    Web3.toTask
+        { func = "eth.getBalance"
+        , args = Encode.list [ Encode.string address ]
+        , expect = expectBigInt
+        , callType = Async
+        }
+
+
+getStorageAt : Address -> Int -> Task Error Hex
+getStorageAt =
+    getStorageAtBlock Latest
+
+
+getStorageAtBlock : BlockId -> Address -> Int -> Task Error Hex
+getStorageAtBlock blockId address position =
+    Web3.toTask
+        { func = "eth.getStorageAt"
+        , args = Encode.list [ Encode.string address, Encode.int position, getBlockIdValue blockId ]
+        , expect = expectString
+        , callType = Async
+        }
+
+
+getCode : Address -> Task Error Bytes
+getCode =
+    getCodeAtBlock Latest
+
+
+getCodeAtBlock : BlockId -> Address -> Task Error Bytes
+getCodeAtBlock blockId address =
+    Web3.toTask
+        { func = "eth.getStorageAt"
+        , args = Encode.list [ Encode.string address, getBlockIdValue blockId ]
+        , expect = expectString
+        , callType = Async
+        }
+
+
+getBlock : BlockId -> Task Error Block
+getBlock blockId =
     Web3.toTask
         { func = "eth.getBlock"
-        , args = Encode.list [ Encode.int blockNum ]
+        , args = Encode.list [ getBlockIdValue blockId, Encode.bool False ]
         , expect = expectJson blockDecoder
         , callType = Async
         }
 
 
-estimateGas : TxParams -> Task Error Int
-estimateGas txParams =
+
+-- TODO Change name of BlockTxObjs type?
```

Comment:
We can get a block with the list of transactions as TxHashes or TxObjs. There is a function for each one.

We have two different type aliases and decoders: Block & BlockTxObjs. If we can come up with a more elegant representation, then hooray.